src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpHandler.cs

```diff
@@ -862,13 +862,14 @@ private void EnsureSessionHandleExists(WinHttpRequestState state)
 
                 state.CancellationToken.ThrowIfCancellationRequested();
 
-                responseMessage = WinHttpResponseParser.CreateResponseMessage(state, _doManualDecompressionCheck);
-
                 // Since the headers have been read, set the "receive" timeout to be based on each read
                 // call of the response body data. WINHTTP_OPTION_RECEIVE_TIMEOUT sets a timeout on each
                 // lower layer winsock read.
                 uint optionData = (uint)_receiveDataTimeout.TotalMilliseconds;
+                Debug.Assert(state.RequestHandle != null);
```

Comment:
Alternatively you could add this as an assert to SetWinHttpOption, to ensure that the handle is never null when passed in.